irrep/kpoint.py

```diff
@@ -123,111 +112,112 @@ class Kpoint:
     ):
         self.spinor = spinor
         self.ik0 = ik + 1  # the index in the WAVECAR (count start from 1)
-        self.Nband = IBend - IBstart
-        #        self.n=np.arange(IBstart,IBend)+1
+        self.num_bands = num_bands
         self.RecLattice = RecLattice
-        self.symmetries_SG = symmetries_SG  # lazy_property needs it
+        self.upper = upper
 
-        if code.lower() == "vasp":
-            self.WF, self.ig = self.__init_vasp(
-                WCF, ik, NBin, IBstart, IBend, Ecut, Ecut0
-            )
-        elif code.lower() == "abinit":
-            self.WF, self.ig = self.__init_abinit(
-                fWFK,
-                ik,
-                NBin,
-                IBstart,
-                IBend,
-                Ecut,
-                Ecut0,
-                kpt=kpt,
-                npw_=npw_,
-                flag=flag,
-                usepaw=usepaw,
-            )
-        elif code.lower() == "espresso":
-            self.WF, self.ig = self.__init_espresso(
-                prefix, ik, IBstart, IBend, Ecut, Ecut0, kptxml=kptxml,
-                spin_channel=spin_channel,IBstartE=IBstartE
-            )
-        elif code.lower() == "wannier":
-            self.WF, self.ig = self.__init_wannier(
-                NBin, IBstart, IBend, Ecut, kpt=kpt, eigenval=eigenval
-            )
-        else:
-            raise RuntimeError("unknown code : {}".format(code))
+        self.k = kpt
+        self.WF = WF
+        self.Energy = Energy
+        self.ig = ig
+        self.upper = upper
 
+        self.k_refUC = np.dot(refUC.T, self.k)
         self.WF /= (
             np.sqrt(np.abs(np.einsum("ij,ij->i", self.WF.conj(), self.WF)))
-        ).reshape(self.Nband, 1)
+        ).reshape(self.num_bands, 1)
 
-    def copy_sub(self, E, WF):
+        # Determine little group and keep only passed symmetries
+        if symmetries is None:
+            symmetries = [symop.ind for symop in symmetries_SG]
+        self.little_group = []
+        for symop in symmetries_SG:
+            if symop.ind not in symmetries:
+                continue
+            k_rotated = np.dot(np.linalg.inv(symop.rotation).T, self.k)
+            dkpt = np.array(np.round(k_rotated - self.k), dtype=int)
+            if np.allclose(dkpt, k_rotated - self.k):
+                self.little_group.append(symop)
+
+        # Sort symmetries based on their indices
+        argsort = np.argsort([symop.ind for symop in self.little_group])
+        self.little_group = [self.little_group[ind] for ind in argsort]
+
+        # Determine degeneracies
+        self.borders = np.hstack([
+             [0],
+             np.where(self.Energy[1:] - self.Energy[:-1] > degen_thresh)[0] + 1,
+             [self.num_bands],
+        ])
+        self.degeneracies = self.borders[1:] - self.borders[:-1]
+
+        # Calculate traces
+        self.char, self.char_refUC, self.Energy = self.calculate_traces(refUC, shiftUC, symmetries_tables, degen_thresh)
```

Comment:
Do we need to calculate traces, in case of unfolding? 
I guess, we need a flag to disable it this may be related to the longer evaluation time in mentioned in #73

Reply:
That's a good point. I have added the argument `traces` for the  `BandStructure` and `Kpoint` classes. It is a boolean (`True` by default) that if set to `False` prevents the calculation of traces. Version labeled as 1.9.3.

Reply:
I see that you have implemented this option in the branch fix-issue-73 of the official repository, motivated by #73 , so it is done.

How should we combine that branch and this pull request?

Reply:
I think you may just merge that branch into yours. 

If you want, you may create a PR from that branch to your branch, and accept it yourself (since it is on your repository)

Reply:
perfect, I have done so. It is now implemented in this PR.